internal/tableland/impl/mesa.go

```diff
@@ -219,6 +224,11 @@ func (t *TablelandMesa) SetController(
 		return tableland.SetControllerResponse{}, fmt.Errorf("chain id %d isn't supported in the validator", chainID)
 	}
 
+	if !stack.AllowTransactionRelay {
```

Comment:
Blocks `SetController` call on a specific chain if the flag is set to false on that chain